discovery-provider/src/tasks/aggregates/__init__.py

```diff
@@ -20,13 +21,19 @@ def init_task_and_acquire_lock(
     db,
     redis,
     table_name,
-    aggregate_func,
+    aggregate_func: Callable[[Session, Redis], Any],
     timeout=60 * 10,
     blocking_timeout=None,
     lock_name=None,
 ):
+
+    current_frame = currentframe()
     # get name of the caller function
-    task_name = currentframe().f_back.f_code.co_name
+    task_name = (
+        current_frame.f_back.f_code.co_name
+        if current_frame is not None and current_frame.f_back is not None
+        else "unknown"
+    )
```

Comment:
This part of the change was due to adding types, as `current_frame` and `f_back` are typed as being `Optional`.